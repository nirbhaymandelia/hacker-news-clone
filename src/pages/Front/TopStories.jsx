/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Listing from '../../components/composite/Listing/Listing';
import Pagination from '../../components/composite/Pagination/Pagination';

class TopStories extends Component {
  componentDidMount() {
    const { params } = this.props.match;
    params.page = params.page || 0;
    this.props.fetchTopStories(params);
  }

  componentDidUpdate(prev) {
    const { params } = this.props.match;
    const { params: prevParams } = prev.match;
    if (prevParams.page !== params.page) {
      this.props.fetchTopStories(params);
    }
  }

  render() {
    const { items, loader, current, total } = this.props;
    return (
      <div>
        <Listing items={items} loader={loader} />
        <Pagination current={current} total={total} target="/top" />
      </div>
    );
  }
}

TopStories.propTypes = {
  items: PropTypes.array.isRequired,
  fetchTopStories: PropTypes.func.isRequired,
  loader: PropTypes.bool.isRequired,
  current: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
  match: PropTypes.object.isRequired,
};

export default TopStories;
