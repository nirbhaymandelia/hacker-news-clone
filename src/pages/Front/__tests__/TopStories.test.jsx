/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';
import TopStories from '../TopStories';

describe('TopStories.jsx', () => {
  it('renders correctly', () => {
    const props = {
      items: [],
      fetchTopStories: () => {},
      loader: false,
      current: 0,
      total: 10,
      match: { params: {} },
    };
    const tree = renderer.create(
      <StaticRouter location="someLocation">
        <TopStories {...props} />
      </StaticRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
