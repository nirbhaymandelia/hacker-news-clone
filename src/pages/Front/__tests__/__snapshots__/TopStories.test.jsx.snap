// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopStories.jsx renders correctly 1`] = `
<main
  className="main"
  id="main"
>
  <div
    className="listing"
  />
  <div
    className="pagination py1 pl2"
  >
    <a
      className="primary-large moreBtn"
      href="/top/1"
      onClick={[Function]}
      title="Page 1 of 10"
    >
      More
    </a>
  </div>
</main>
`;
